spi/src/main/java/org/jboss/pnc/spi/datastore/repositories/ProductMilestoneRepository.java

```diff
@@ -18,12 +18,18 @@
 package org.jboss.pnc.spi.datastore.repositories;
 
 import org.jboss.pnc.model.ProductMilestone;
+import org.jboss.pnc.spi.datastore.repositories.api.PageInfo;
 import org.jboss.pnc.spi.datastore.repositories.api.Repository;
 
+import javax.persistence.Tuple;
+import java.util.List;
+
 /**
  * Interface for manipulating {@link org.jboss.pnc.model.ProductMilestone} entity.
  */
 public interface ProductMilestoneRepository extends Repository<ProductMilestone, Integer> {
 
     long countBuiltArtifactsInMilestone(Integer id);
+
+    List<Tuple> getArtifactsDeliveredInMilestones(List<Integer> milestoneIds);
```

Comment:
Describe what's in the tuple and the query you're making. From interface standpoint this looks extremely generic and there's little indication what's happening unless you dig into implementation since this is not the simplest of queries.